kymatio/scattering2d/frontend/base_frontend.py

```diff
@@ -17,24 +17,34 @@ def __init__(self, J, shape, L=8, max_order=2, pre_pad=False,
         self.out_type = out_type
 
     def build(self):
-        self.M, self.N = self.shape
+        M, N = self.shape
 
-        if 2 ** self.J > self.M or 2 ** self.J > self.N:
+        if 2 ** self.J > M or 2 ** self.J > N:
             raise RuntimeError('The smallest dimension should be larger than 2^J.')
-        self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
```

Comment:
Actually, I'm unsure why we do this change here. In the parametric scattering code we access `self.M_padded`, `self.N_padded`... does this need to have the underscore added to it?

Reply:
the idea is that the user shouldn't attempt to change these values in the `Scattering2D` object after it's being constructed. So we put a leading underscore to them to signify that they are "privarte" (although in Python nothing is truly private and there's nothing preventing the user to get or set leading-underscore values)

You might say that the same could be said about `J`, `L`, etc. which really should be stored as `_J`, `_L`, etc. and accessed as properties, but one API change at a time.